azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/domain/IpConfiguration.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.azurecomputearm.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class IpConfiguration{
+
+    @AutoValue
+    public abstract static class IpConfigurationProperties{
+
+        public IpConfigurationProperties() {
+        }// For AutoValue only!
+
+        @Nullable
+        public abstract String provisioningState();
```

Comment:
For properties that are not added here like `publicIPAddress`, `loadBalancerBackendAddressPools`, `loadBalancerInboundNatRules`, please make sure they are optional. If so, let's add a low-priority card on Trello.